spring-rabbit/src/main/java/org/springframework/amqp/rabbit/core/RabbitOperations.java

```diff
@@ -42,6 +42,20 @@
 	 */
 	<T> T execute(ChannelCallback<T> action) throws AmqpException;
 
+	/**
+	 * Execute the callback with a channel and reliably close the channel afterwards.
+	 * @param action the call back.
+	 * @param publisherConnectionIfPossible true to use a separate publisher connection if possible.
+	 * @param <T> the return type.
+	 * @return the result from the
+	 * {@link ChannelCallback#doInRabbit(com.rabbitmq.client.Channel)}.
+	 * @throws AmqpException if one occurs.
+	 * @since 2.0.2
+	 */
+	default <T> T execute(ChannelCallback<T> action, boolean publisherConnectionIfPossible) throws AmqpException {
```

Comment:
Why it is still so overcomplicated?
Why there is just not enough to extract the publisherCF in the `RabbitTemplate.afterPropertiesSet()` and everything rest will remain the same as before?

No? Any explanations, please?

Reply:
Good catch. This is an artifact of some earlier refactoring; it can be removed now.

Reply:
Good catch. This is an artifact of some earlier refactoring; it can be removed now. I have to step out for a while; I'll push another commit tonight.